### ISSUE TITLE
Sample tests failing with Hugo 0.55 to 0.59 (additional p tags and invalid HTML) 

### ISSUE THREAD

**Issue by jaaufauvre** [manixate/jest-hugo#21]
Reproduction steps:

1. Checkout tag 1.0.2
2. Run:
```
npm install
cd demo
npm install
npm run jest
```

Hugo version:
```
Hugo Static Site Generator v0.55.6-A5D4C82D windows/amd64 BuildDate: 2019-05-18T07:57:00Z
```

HTML output (invalid tag sequence):

```html
<p><test name="should render successful with alert"></p>

<p><div>
  <div class="callout callout-alert" role="callout">
    <div class="callout-header">Alert:
    </div>
    <div class="callout-content">An alert type callout</div>
  </div>
</div>
</test></p>
```

Test output:
```
  ● callout › should render successful with alert

    expect(received).toMatchSnapshot()

    Snapshot name: `callout should render successful with alert 1`

    - Snapshot  - 0
    + Received  + 3

    @@ -1,5 +1,8 @@
    + </p>
    +
    + <p>
      <div>
        <div class="callout callout-alert"
          role="callout">
          <div class="callout-header">Alert:
          </div>
```

**Comment by jaaufauvre** [comment]
It seems wrapping the tests into a `div` fixes the issue:

```html
<div>
<test name="should render successful with alert">
  {{% callout type="alert" %}}
  An alert type callout
  {{% /callout %}}
</test>
</div>
``` 

Using `div` instead of `test` tags for defining tests seem to work too:

```html
<div type="test" name="should render successful with alert">
  {{% callout type="alert" %}}
  An alert type callout
  {{% /callout %}}
</div>
```

**Comment by manixate** [comment]
Different Hugo versions behave differently in the output. As a start we can mention that the demo is for a specific version.

On the note of using a `div` instead of `test`, it is possible to do but I picked a custom tag to make it clear that it is only for testing and to ensure that it doesn't interfere with the real code. We can enforce some other attributes if we want to use a `div` example `data-type="test"` but am not sure if it is worth the change.

**Comment by jaaufauvre** [comment]
Hi @manixate, I did a few tests, and `<test />` works with versions of Hugo >= [v0.60.0](https://github.com/gohugoio/hugo/releases/tag/v0.60.0).

For that reason, I would:
* Stick with `<test />` being the only way of defining tests
* Document the fact the test cases must be wrapped into a `<div />` if Hugo version is < [v0.60.0](https://github.com/gohugoio/hugo/releases/tag/v0.60.0) (which hopefully aren't used anymore 😀).

What do you think?

I'm going to open a PR for that. The good news is wrapping the `<test />` elements into a `<div />` will fix the demo for Hugo v0.55 to v0.59 without breaking newer versions.

**Comment by manixate** [comment]
Looks good! Thanks for the PR.
